### PR TITLE
Hostlib

### PR DESCRIPTION
**Pull request by ChrisPattison**
Object oriented interface for SDPB to depend against

### CODE REVIEW

File: interface/Apfp.hpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#pragma once
+#include <hlslib/xilinx/OpenCL.h>
+
+#include <optional>
```
Comment (definelicht) [inline interface/Apfp.hpp]: What is this for?

**Comment by definelicht** [comment]
The rest of the source code uses `.h` instead of `.hpp`. I have no strong feelings either way, but we should be consistent :-)

**Comment by definelicht** [comment]
Remember to clang-format files before pushing!

**Comment by definelicht** [comment]
This interface looks fine to me! Make some tests for the different operations.

File: interface/Apfp.hpp (hunk 1 above)
Comment (ChrisPattison) [reply interface/Apfp.hpp]: Apfp has an hlslib Program wrapped in an optional to allow it to be default constructed. The idea is that later when we add the runtime precision changing we'll have to do some extra processing in the constructor before constructing it

**Comment by ChrisPattison** [comment]
> This interface looks fine to me! Make some tests for the different operations.

Would it make sense for some of the library tests to go through this interface instead?

**Comment by definelicht** [comment]
> Would it make sense for some of the library tests to go through this interface instead?

Why not both? ;-)

**Pull request merged**